### ISSUE TITLE
PayloadTooLargeError: request entity too large

### ISSUE THREAD

**Issue by JannikZed** [fission/fission#1132]
We are actively using the mqt-trigger system with fission and everything is running well (normally). Today I was manually looking around the function execution pods of fission and saw a super large amount of errors like this: 
```
PayloadTooLargeError: request entity too large
at readStream (/usr/src/app/node_modules/raw-body/index.js:155:17)
at getRawBody (/usr/src/app/node_modules/raw-body/index.js:108:12)
at read (/usr/src/app/node_modules/body-parser/lib/read.js:77:3)
at jsonParser (/usr/src/app/node_modules/body-parser/lib/types/json.js:135:5)
at Layer.handle [as handle_request] (/usr/src/app/node_modules/express/lib/router/layer.js:95:5)
at trim_prefix (/usr/src/app/node_modules/express/lib/router/index.js:317:13)
at /usr/src/app/node_modules/express/lib/router/index.js:284:7
at Function.process_params (/usr/src/app/node_modules/express/lib/router/index.js:335:12)
at next (/usr/src/app/node_modules/express/lib/router/index.js:275:10)
at urlencodedParser (/usr/src/app/node_modules/body-parser/lib/types/urlencoded.js:100:7)
::ffff:10.42.5.126 - - [27/Mar/2019:11:55:19 +0000] "POST / HTTP/1.1" 413 1109 "-" "Go-http-client/1.1"
PayloadTooLargeError: request entity too large
at readStream (/usr/src/app/node_modules/raw-body/index.js:155:17)
at getRawBody (/usr/src/app/node_modules/raw-body/index.js:108:12)
at read (/usr/src/app/node_modules/body-parser/lib/read.js:77:3)
at jsonParser (/usr/src/app/node_modules/body-parser/lib/types/json.js:135:5)
at Layer.handle [as handle_request] (/usr/src/app/node_modules/express/lib/router/layer.js:95:5)
at trim_prefix (/usr/src/app/node_modules/express/lib/router/index.js:317:13)
at /usr/src/app/node_modules/express/lib/router/index.js:284:7
at Function.process_params (/usr/src/app/node_modules/express/lib/router/index.js:335:12)
at next (/usr/src/app/node_modules/express/lib/router/index.js:275:10)
at urlencodedParser (/usr/src/app/node_modules/body-parser/lib/types/urlencoded.js:100:7)
::ffff:10.42.5.126 - - [27/Mar/2019:11:55:19 +0000] "POST / HTTP/1.1" 413 1109 "-" "Go-http-client/1.1"
``` 

And that's really all the information I have.. I just upgraded fission to the newset version 1.1.0 but the problem still exists. We are already having a big amount of services writing to an input NATS topic, so I also don't know which of the services produce this problem .. the fission error message is not helping debugging here :/ 

Did anybody came across the same issue, too?

**Comment by life1347** [comment]
Hi @JannikZed , sorry for the late reply. It's the first time for me to see such error log. I'm not very familiar with Node, but looks like it may be able to solve by adding express (https://stackoverflow.com/questions/19917401/error-request-entity-too-large).  As a workaround you need to modify the server.js ([here](https://github.com/fission/fission/tree/master/environments/nodejs)) and rebuild nodejs container image. Let me know wether it works or not.

If it still not work, would you mind to dump fission service log with commands and share the output? Thanks
```
$ kubectl -n fission logs -f <mqtrigger-nats-streaming pod>
$ kubectl -n fission logs -f <nats-streaming pod>
```

**Comment by JannikZed** [comment]
Hi there,
I tried it out and it seems to work. Do you want it included in the main repo? I can create a pull request, or is there any possible problem with a higher body-size? I don't know, why we are the first to run into this issue actually.
Here is my updated docker version: https://cloud.docker.com/u/jannikz/repository/docker/jannikz/fission-limit-test

**Comment by life1347** [comment]
Hey @JannikZed good to hear it works now. Do you know how large the request body is? 

In your case, the component through the call path may be affected (router, mqtrigger and NATS-streaming) and consume more memory cause they have to hold request content till the end of the call.

It's a good idea to open a PR! At that we may able to do some extra tests to see the impact  of the changes for Node (and test of other environments of course). 

P.S I got HTTP 500 for the dockerhub link, maybe it's private?

**Comment by JannikZed** [comment]
seems like NATS got a maximum size of 1mb: https://nats.io/documentation/faq/#msgsize
So I will create a pull request with the max body-size to be also 1mb - this is already a lot bigger than the standard body-size.
The docker-image should be accessible via `docker pull jannikz/fission-limit-test` - it's public.

**Comment by life1347** [comment]
Fixed by #1149
